### DISCUSSION TITLE
DatePicker - Localization for Japanese lang

### DISCUSSION THREAD

**Discussion by makarandkeer** [adobe/react-spectrum#3303] [General]
We are using DatePicker and for Japanese language, its showing the Calendar as below. However it appends `日` to each day which seems to be unexpected behavior. Is this is known issue and if yes is there workaround to fix it? 

![image](https://user-images.githubusercontent.com/32228809/178618164-ee25279b-059c-477c-a1f0-ab35efd54895.png)

**Comment by devongovett** [comment]
hmm, we don't do our own formatting, this is what is returned from the `Intl.DateTimeFormat` API. For example:

```
new Intl.DateTimeFormat('ja-JP', {day: 'numeric'}).format(new Date())
```

**Reply by nozokada** [reply]
I think `Intl.DateTimeFormat` should be used for formatting a date and time but not for a calendar day. It looks weird in Japanese at least. Are there any other languages that take advantage of it?

**Comment by mortnod** [comment]
We have a similar problem for scandinavian locales like `nb-NO` (Norwegian) and `da-DK` (Danish). Every date cell ends up having a dot which should not be there. 

While not egregiously wrong, it'd be like having an American calendar where every cell said stuff like "1st, 2nd, 3rd, 4th".

<img width="349" alt="Screenshot 2023-02-06 at 10 30 55" src="https://user-images.githubusercontent.com/3471625/216936572-8c875a5f-11a7-4ae3-8a94-6e516f20d0e1.png">

Would love for a way to override this behaviour. Thanks for the amazing project in general ✌️

**Reply by lreis** [reply]
Hi @devongovett 

The formatting of these cell contents seem to be handled at 

https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/calendar/src/useCalendarCell.ts#L305

How about using the formatToParts method to capture the numeric value only:

`let formattedDate = useMemo(() => cellDateFormatter.formatToParts(nativeDate)[0].value, [cellDateFormatter, nativeDate]);`

**Comment by lreis** [comment]
I opened an issue for this, see https://github.com/adobe/react-spectrum/issues/4135
